src/tools/rbd_wnbd/wnbd_handler.cc

```diff
@@ -367,6 +369,28 @@ void WnbdHandler::LogMessage(
           << WnbdLogLevelToStr(LogLevel) << " " << Message << dendl;
 }
 
+int WnbdHandler::resize(uint64_t new_size)
+{
+  int err = 0;
+  
+  uint64_t new_block_count = new_size / block_size;
+
+  dout(5) << "Resizing disk. Block size: " << block_size
+          << ". New block count: " << new_block_count
+          << ". Old block count: "
+          << wnbd_disk->Properties.BlockCount << "." << dendl;
+  err = WnbdSetDiskSize(wnbd_disk, new_block_count);
+
+  if (err) {
+    derr << "WNBD: Setting disk size failed with error: "
+         << err << dendl;
+    return -EINVAL;
+  }
+
+  dout(5) << "Successfully resized disk to: " << new_block_count << " blocks" << dendl;
+
+  return err;
```

Comment:
Since this is a success path, it's better to return 0 directly.  Also another excessive blank line:
```suggestion
  return 0;
```